cmd/utils/flags.go

```diff
@@ -774,11 +777,11 @@ var (
 
 // MakeDataDir retrieves the currently requested data directory, terminating
 // if none (or the empty string) is specified. If the node is starting a testnet,
-// the a subdirectory of the specified datadir will be used.
+// then a subdirectory of the specified datadir will be used.
 func MakeDataDir(ctx *cli.Context) string {
 	if path := ctx.GlobalString(DataDirFlag.Name); path != "" {
-		if ctx.GlobalBool(TestnetFlag.Name) {
-			return filepath.Join(path, "testnet")
+		if ctx.GlobalBool(LegacyTestnetFlag.Name) || ctx.GlobalBool(RopstenFlag.Name) {
+			return filepath.Join(path, "ropsten")
```

Comment:
Plopp. This is where you forgot to add the legacy `testnet` folder check.

Reply:
Ok. I was assuming that `MakeDataDir()` is only called when _creating_ a new datadir? we don't want to create new `testnet` subdirs, do we? Is there any instance where this is called for an existing datadir?

Reply:
When you create a `console.Config`, this method is used. That will make the console believe it's in a different folder than the rest of Geth.

Reply:
https://github.com/ethereum/go-ethereum/search?q=%22console.Config%22&unscoped_q=%22console.Config%22 

3 instances in this file

Reply:
Btw, it does not create a datadir. The name is a bit misleading. All the methods that take the flags and "Make" some internal config out of them are called "MakeXXX". I guess this is more meaningful for "MakeEthConfig", but it just got used as a general style for all.